python/lsst/sconsUtils/state.py

```diff
@@ -329,40 +351,52 @@ def ClassifyCc(context):
     if env.GetOption("clean") or env.GetOption("no_exec") or env.GetOption("help"):
         env.whichCc = "unknown"         # who cares? We're cleaning/not execing, not building
     else:
-        if env['cc'] != '':
-            CC = CXX = None
-            if re.search(r"^gcc(-\d+(\.\d+)*)?( |$)", env['cc']):
-                CC = env['cc']
-                CXX = re.sub(r"^gcc", "g++", CC)
-            elif re.search(r"^icc( |$)", env['cc']):
-                CC = env['cc']
-                CXX = re.sub(r"^icc", "icpc", CC)
-            elif re.search(r"^clang( |$)", env['cc']):
-                CC = env['cc']
-                CXX = re.sub(r"^clang", "clang++", CC)
-            elif re.search(r"^cc( |$)", env['cc']):
-                CC = env['cc']
-                CXX = re.sub(r"^cc", "c++", CC)
-            else:
-                log.fail("Unrecognised compiler:%s" % env['cc'])
-            env0 = SCons.Script.Environment()
-            if CC and env['CC'] == env0['CC']:
-                env['CC'] = CC
-            if CC and env['CXX'] == env0['CXX']:
-                env['CXX'] = CXX
-        conf = env.Configure(custom_tests={'ClassifyCc': ClassifyCc})
-        env.whichCc, env.ccVersion = conf.ClassifyCc()
-
-        # If we have picked up a default compiler called gcc that is really
-        # clang, we call it clang to avoid confusion (gcc on macOS has subtly
-        # different options)
-        if not env['cc'] and env.whichCc == "clang" and env['CC'] == "gcc":
-            env['CC'] = "clang"
-            env['CXX'] = "clang++"
+        if 'SCONSUTILS_USE_CONDA_COMPILERS' in os.environ:
+            # conda-build expects you to use the compilers as-is
+            env['CC'] = os.environ['CC']
+            env['CXX'] = os.environ['CXX']
+
+            conf = env.Configure(custom_tests={'ClassifyCc': ClassifyCc})
+            env.whichCc, env.ccVersion = conf.ClassifyCc()
+            if not env.GetOption("no_progress"):
+                log.info("CC is **CONDA** %s version %s" % (env.whichCc, env.ccVersion))
+            conf.Finish()
+        else:
+            if env['cc'] != '':
+                CC = CXX = None
+                if re.search(r"^gcc(-\d+(\.\d+)*)?( |$)", env['cc']):
+                    CC = env['cc']
+                    CXX = re.sub(r"^gcc", "g++", CC)
+                elif re.search(r"^icc( |$)", env['cc']):
+                    CC = env['cc']
+                    CXX = re.sub(r"^icc", "icpc", CC)
+                elif re.search(r"^clang( |$)", env['cc']):
+                    CC = env['cc']
+                    CXX = re.sub(r"^clang", "clang++", CC)
+                elif re.search(r"^cc( |$)", env['cc']):
+                    CC = env['cc']
+                    CXX = re.sub(r"^cc", "c++", CC)
+                else:
```

Comment:
Originally, I just had something like:

```suggestion
                elif env['CXX'] != '':
                    CC = env['CC']
                    CXX = env['CXX']
                    log.warn(f"Unrecognized compilers: CC={CC}, CXX={CXX}")
                else:
```

Which worked fine, especially since the classify code still ran later, but this seems more complete.

Reply:
Right. I was trying to be more explicit so that changes in one code path don't break the other. Once we know we are inside conda build, there is no reason to guess. Conda's compilers expect you to respect the standard compiler env vars absolutely.